src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/config/Config.java

```diff
@@ -378,24 +426,34 @@ public void setValues(JSONObject formData) {
                 "projectListFetchDelay",
                 DEFAULT_PROJECT_LIST_FETCH_DELAY);
 
-        projectListRefreshInterval = formData.optInt(
-                "projectListRefreshInterval",
-                DEFAULT_PROJECT_LIST_REFRESH_INTERVAL);
-        enableProjectAutoCompletion = formData.optBoolean(
-                "enableProjectAutoCompletion",
-                DEFAULT_ENABLE_PROJECT_AUTO_COMPLETION);
-
         categories = new LinkedList<VerdictCategory>();
         if (formData.has("verdictCategories")) {
             Object cat = formData.get("verdictCategories");
             if (cat instanceof JSONArray) {
-                for (Object jsonObject : (JSONArray)cat) {
-                    categories.add(VerdictCategory.createVerdictCategoryFromJSON((JSONObject)jsonObject));
+                for (Object jsonObject : (JSONArray) cat) {
+                    categories.add(VerdictCategory.fromJSON((JSONObject) jsonObject, formData));
                 }
-            } else if (cat instanceof JSONObject) {
-                categories.add(VerdictCategory.createVerdictCategoryFromJSON((JSONObject)cat));
+            }
+            else if (cat instanceof JSONObject) {
+                categories.add(VerdictCategory.fromJSON((JSONObject) cat, formData));
             }
         }
+        assertDefaultCategories();
+        //cover the migration of default gerrit reporting values
+        for (VerdictCategory cat : categories) {
+            if (formData.has(cat.getVerdictValue() + "Successful"))
```

Comment:
Isn't this code already in `VerdictCategory.fromJSON`?

Reply:
Not quite, in VerdictCategory we create a new object with default reporting values, whereas this section is to cover the case when we switch from a config that doesn't support multiple labels to one that does.